spec/registry.md

```diff
@@ -255,3 +255,548 @@ which only expects the `accord` option:
 > .input {$object :noun case=nominative}
 > {{You see {$color :adjective article=indefinite accord=$object} {$object}!}}
 >```
+
+# Default Registry
+
+This section describes the functions which each implementation MUST provide to be conformant with
+this specification.
+
+## String Value Selection and Formatting
+
+### Functions
+
+The following functions are provided:
+
+The function `:string` provides string selection and formatting.
+
+### Operands
+
+The _operand_ is any literal or an implementation-defined set of string or
```

Comment:
I wonder if we should use the term "string" or "string literal" here (and include a definition of "string literal" as an arbitrary string which the formatting function may parse in any way) to avoid confusion with `Literal` in the grammar.

Reply:
Actually, this text really **_means_** the production `Literal` in the grammar. The operand is any such literal _or_ implementation defined types.

Reply:
What's confusing me is that there are several different meanings of "literal":

* any string
* a data model node representing a `Literal` node in the tree
* a string that is generated by one of the following three productions in the grammar: `*(quoted-char / quoted-escape)`; `name` or `number-literal`

Put differently, the string "|1|" is a valid syntactic representation of a number literal, but not a valid argument to the number formatter.

Reply:
`{|1| :number}` is valid syntax. But the implementation parses `|1|` into a `Literal` data model node (or does something observably the same), and is then responsible for extracting the string "1" from that node and passing it into the runtime function that implements `:number`. If it passed the string `|1|` to `:number` instead, that wouldn't be spec-compliant.

Reply:
I think we're saying the same thing? The literal value of `|1|` is the string `1` and doesn't include the quoting pipe symbols. Does my suggestion help express that? What should I change?

Reply:
See the suggestion I made (at the same time as you wrote this comment :)